Pokedex/ListViewController.swift

```diff
@@ -0,0 +1,340 @@
+            let vc = segue.destination as? ProfileViewController
+            vc?.passedPokemon = selectedPokemon
+        }
+    }
+    
+}
+
+extension ListViewController: UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout{
+   
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return 20
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "pokemonCell", for: indexPath) as! PokemonCollectionViewCell
+        for subview in cell.contentView.subviews{
+            subview.removeFromSuperview()
+        }
+        cell.awakeFromNib()
+        return cell
+
+    }
+    func collectionView(_ collectionView: UICollectionView, willDisplay cell: UICollectionViewCell, forItemAt indexPath: IndexPath) {
+        let pokemonCell = cell as! PokemonCollectionViewCell
+        let pokemonDisplayed = randomPokemon[indexPath.row]
+        
+        //for label
+        pokemonCell.pokemonNameLabel.text = pokemonDisplayed.name
+        var name:String? = pokemonDisplayed.name
+            //this makes the cells look nicer as it cuts out the pokemon's nickname if it has one
+        if pokemonDisplayed.name.contains("("){
+            let index = name!.index(of: "(")
+            name = name?.substring(to: index!)
+            
+        }
+        pokemonCell.pokemonNameLabel.text = name!
+
+        
+        //for image
+        //checks if imageurl is nil
+        var imgURL: URL
+        if let _ = Foundation.URL(string: pokemonDisplayed.imageUrl){
+            imgURL = Foundation.URL(string: pokemonDisplayed.imageUrl)!
+            
+            //This is to check if the url actually downloads an image (checks for nil)
+            let placeholderImageView: UIImageView? = UIImageView()
+            placeholderImageView?.downloadedFrom(url: imgURL)
+            
+            if placeholderImageView != nil{
+                pokemonCell.pokemonImageView?.downloadedFrom(url: imgURL)
+            }else{
+                let imageName = "noimage"
+                let image = UIImage(named: imageName)
+                pokemonCell.pokemonImageView? = UIImageView(image: image!)
+                
+            }
+            
+            
+        }else{
+            let imageName = "noimage"
+            let image = UIImage(named: imageName)
+            pokemonCell.pokemonImageView? = UIImageView(image: image!)
+        }
+        
+      
+
+        
+        
+    }
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width:((view.frame.width)/2) - 2, height: 200)
```

Comment:
same comment as before, use UIScreen bounds instead of view frame dimensions. It might be okay to use view frame dimensions if those are determined by UIScreen when initialized however.